Test/src/FormatStyle_unitest.cpp

```diff
@@ -1560,4 +1560,115 @@ local t =
 }
 )",
             style));
-}
\ No newline at end of file
+}
+
+TEST(FormatByStyleOption, end_statement_with_semicolon) {
+    LuaStyle style;
+
+    style.end_statement_with_semicolon = EndStmtWithSemicolon::Keep;
+    EXPECT_TRUE(TestHelper::TestFormatted(
+            R"(local func = testFunc; print(type(nil)); -- nil;
+print("Hello"); print(" "); --[[ test comment --]] print("World")
+for i, v in ipairs(values) do
+    if v then
+        print(i, v, "True");
+    else
+        print(i, v, "False")
+    end;
+end;
+local table1 = { 1, 2, 3 }
+)",
+            R"(local func = testFunc; print(type(nil)); -- nil;
+print("Hello"); print(" "); --[[ test comment --]] print("World")
+for i, v in ipairs(values) do
+    if v then
+        print(i, v, "True");
+    else
+        print(i, v, "False")
+    end;
+end;
+local table1 = { 1, 2, 3 }
+)",
+            style));
+
+    style.end_statement_with_semicolon = EndStmtWithSemicolon::Always;
+    EXPECT_TRUE(TestHelper::TestFormatted(
+            R"(local func = testFunc; print(type(nil)); -- nil;
+print("Hello"); print(" "); --[[ test comment --]] print("World")
+for i, v in ipairs(values) do
+    if v then
+        print(i, v, "True");
+    else
+        print(i, v, "False")
+    end;
+end;
+local table1 = { 1, 2, 3 }
+)",
+            R"(local func = testFunc; print(type(nil)); -- nil;
+print("Hello"); print(" "); --[[ test comment --]] print("World");
+for i, v in ipairs(values) do
+    if v then
+        print(i, v, "True");
+    else
+        print(i, v, "False");
+    end;
+end;
+local table1 = { 1, 2, 3 };
+)",
+            style));
+
+    style.end_statement_with_semicolon = EndStmtWithSemicolon::Never;
+    EXPECT_TRUE(TestHelper::TestFormatted(
+            R"(local func = testFunc; print(type(nil)); -- nil;
+print("Hello"); print(" "); --[[ test comment --]] print("World")
+for i, v in ipairs(values) do
+    if v then
+        print(i, v, "True");
+    else
+        print(i, v, "False")
+    end;
+end;
+local table1 = { 1, 2, 3 }
+)",
+            R"(local func = testFunc
+print(type(nil))                         -- nil;
```

Comment:
Why so much whitespace?

Reply:
The default formatting configuration (which is used in the tests, apart from the semicolon configuration) seems to want to keep the comment at the same indentation where it was before reformatting, which I assume is handled by another analyzer/formatter.